### PR TITLE
Speaking test module

### PR DESCRIPTION
**Pull request by dimahalaiskiy**

### CODE REVIEW

**Comment by dimahalaiskiy** [comment]
audio-recorder

File: exadel-trainee-app/src/app/models/speaking-test/speaking-test.component.html [hunk 1]
```diff
@@ -1,4 +1,12 @@
-<button mat-raised-button color="primary" (click)="startRecording()">Start recording</button>
-<button mat-raised-button color="warn" (click)="stopRecording()">Stop</button>
-
-<audio *ngFor="let audio of audioFiles" controls="true" [src]="audio.src"></audio>
+<mat-card class=text-of-theme>What are you speaking about?</mat-card>
+<section class="speaking-module-position" ng-controller="controller">
+  <button mat-raised-button class="button-position" [color]="getColor()" (click)="toggleRecording()">
+    {{ isRecording == false ? 'Start Recording' : 'Stop Recording' }}
```
Comment (RomPavel) [inline exadel-trainee-app/src/app/models/speaking-test/speaking-test.component.html:4]: you can simple use **!isRecording ? 'Start Recording' : 'Stop Recording'**

File: exadel-trainee-app/src/app/models/speaking-test/speaking-test.module.ts [hunk 2]
```diff
@@ -2,11 +2,12 @@ import { NgModule } from '@angular/core';
 import { CommonModule } from '@angular/common';
 import { FormsModule } from '@angular/forms';
 import { MatButtonModule } from '@angular/material/button';
+import { MatCardModule } from '@angular/material/card';
 import { SpeakingTestComponent } from './speaking-test.component';
 import { SpeakingTestRoutingModule } from './speaking-test-routing.module';
 
 @NgModule({
-  imports: [SpeakingTestRoutingModule, CommonModule, FormsModule, MatButtonModule],
+  imports: [SpeakingTestRoutingModule, CommonModule, FormsModule, MatButtonModule, MatCardModule],
```
Comment (Inunor) [inline exadel-trainee-app/src/app/models/speaking-test/speaking-test.module.ts:10]: You should use https://github.com/Inunor/internship/blob/dev/exadel-trainee-app/src/app/models/angular-material-common.module.ts. Someone already created it, it exports all the necessary angular material modules

File: exadel-trainee-app/src/app/models/speaking-test/speaking-test.component.ts [hunk 3]
```diff
@@ -36,10 +38,24 @@ export class SpeakingTestComponent implements OnInit {
...
       };
     } catch (err) {
+      // eslint-disable-next-line no-alert
       alert('Error capturing audio.');
     }
   }
 
+  toggleRecording() {
+    this.isRecording = !this.isRecording;
+    if (this.isRecording === true) {
```
Comment (Inunor) [inline exadel-trainee-app/src/app/models/speaking-test/speaking-test.component.ts:48]: You can use just if (this.isRecording)

**Review by Inunor — CHANGES_REQUESTED** [review]

File: exadel-trainee-app/src/app/models/speaking-test/speaking-test.component.ts [hunk 4]
```diff
@@ -36,10 +38,24 @@ export class SpeakingTestComponent implements OnInit {
...
+      this.startRecording();
+    } else {
+      this.stopRecording();
+    }
+  }
+
+  getColor() {
+    return this.isRecording === false ? 'primary' : 'warn';
```
Comment (RomPavel) [inline exadel-trainee-app/src/app/models/speaking-test/speaking-test.component.ts:56]: !this.isRecording

**Review by RomPavel — APPROVED** [review]

**Review by RomPavel — APPROVED** [review]

**Pull request merged**